### PR TITLE
MARKET-1733 [Double Calendar] Add date format selection in settings

### PR DESCRIPTION
**Pull request by Saharij**
UI Builder - https://develop.backendless.com/app/zahar_mychka_components/ui-builder/default/pages/double-calendar

### CODE REVIEW

File: components/bl-double-calendar-component/src/helpers/utils.js [hunk 1]
```diff
@@ -13,3 +13,21 @@ export function differenceInDays(start, end) {
...
+    return null;
+  }
+
+  const splitDate = date.split("-");
+  const { DD, MM, YYYY } = format.split("-").reduce((acc, item, index) => {
+    return {
+      ...acc,
+      [item]: index,
+    };
+  }, {});
```
Comment (v-excelsior) [inline components/bl-double-calendar-component/src/helpers/utils.js:30]: getFormatPartsOrder

File: components/bl-double-calendar-component/src/helpers/utils.js [hunk 2]
```diff
@@ -13,3 +13,21 @@ export function differenceInDays(start, end) {
...
 }
+
+export function normalizeFormat(date, format) {
+  if (!date) {
+    console.warn('Date is not provided!')
+
+    return null;
+  }
+
+  const splitDate = date.split("-");
```
Comment (v-excelsior) [inline components/bl-double-calendar-component/src/helpers/utils.js:24]: splitDate -> dateParts

File: components/bl-double-calendar-component/src/index.js [hunk 3]
```diff
@@ -29,8 +31,8 @@ export default function DoubleCalendarComponent({ component, eventHandlers }) {
         />
         :
         <DateRange
-          toDate={ toDate }
-          fromDate={ fromDate }
+          toDate={ normalizeFormat(toDate, dateFormat) }
```
Comment (v-excelsior) [inline components/bl-double-calendar-component/src/index.js:34]: ```const formatNormalizer = useMemo(()=> date => normalizeFormat(dateFormat, date))```

Change args order and use closures here. Also both ranges could be memoized too.
Contact me if something unclear.

File: components/bl-double-calendar-component/src/helpers/use-actions.js [hunk 4]
```diff
@@ -1,18 +1,23 @@
...
+      const normalizedFromDate = new Date(normalizeFormat(fromDate, dateFormat));
+      const normalizedToDate = new Date(normalizeFormat(toDate, dateFormat));
+
+      const diffInTime = differenceInTime(normalizedFromDate, normalizedToDate);
 
       if (diffInTime >= 0) {
-        setStartDate(new Date(fromDate));
-        setEndDate(new Date(toDate));
+        setStartDate(normalizedFromDate);
+        setEndDate(normalizedToDate);
```
Comment (v-excelsior) [inline components/bl-double-calendar-component/src/helpers/use-actions.js:20]: I assume `normalizedFromDate` and `normalizedToDate` could be passed as argument here, check it please.

File: components/bl-double-calendar-component/src/helpers/use-actions.js [hunk 5]
```diff
@@ -1,18 +1,23 @@
...
       } else {
         console.warn("Date is not valid!");
```
Comment (v-excelsior) [inline components/bl-double-calendar-component/src/helpers/use-actions.js:22]: Please add reason here.

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-double-calendar-component/src/helpers/use-actions.js (hunk 4 above)
Comment (Saharij) [reply components/bl-double-calendar-component/src/helpers/use-actions.js:20]: Dates that passed to this function came from user input, that's why we need to normalize it here. If I understood you correctly.

**Pull request merged**